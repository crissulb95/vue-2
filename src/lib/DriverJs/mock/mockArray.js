export const mockArray = () => [
  {
    element: ".pokemon-page",
    popover: {
      className: "modal-popover-class introduction",
      title: `
      <div class="top-container">
        <figure class="image-container">
          <img src="/Empty state.png" alt="logo safit" />
          <figcaption>logo safit</figcaption>
        </figure>
        <h2>Bienvenido/a al nuevo Safit</h2>
      </div>`,
      description: `
        <p class="text-container">Una manera más eficiente de operar las funcionalidades de antes, junto a nuevos módulos</p>
        <ul class="bullet-points-container">
          <li class="bullet-point active"></li>
          <li class="bullet-point"></li>
        </ul>
      `,
      position: "mid-center",
      closeBtnText: "Omitir",
      nextBtnText: "Siguiente",
      prevBtnText: "",
      doneBtnText: "Finalizar",
    },
  },
  {
    element: ".pokemon-picture-container",
    popover: {
      className: "modal-popover-class introduction",
      title: `
      <div class="top-container">
        <figure class="image-container">
          <img src="/Empty state.png" alt="logo safit" />
          <figcaption>logo safit</figcaption>
        </figure>
        <h2>Hagamos un recorrido</h2>
      </div>`,
      description: `
        <p class="text-container">De manera rápida le enseñaremos algunas novedades del sistema para que pueda utilizarlo de la mejor manera</p>
        <ul class="bullet-points-container">
          <li class="bullet-point"></li>
          <li class="bullet-point active"></li>
        </ul>
      `,
      position: "mid-center",
      closeBtnText: "Omitir",
      nextBtnText: "Iniciar recorrido",
      prevBtnText: "",
      doneBtnText: "Finalizar",
    },
  },
  {
    element: "#replay",
    popover: {
      className: "modal-popover-class final",
      title: "<h2>Manténgase actualizado</h2>",
      description:
        "En la seccion <strong>notificaciones</strong> le informaremos acerca de las tareas realizadas en la plataforma, De manera rápida le enseñaremos algunas novedades del sistema para que pueda utilizarlo de la mejor manera",
      position: "right",
      closeBtnText: "Omitir",
      nextBtnText: "Siguiente",
      prevBtnText: "",
      doneBtnText: "Finalizar",
    },
  },
  /*
  {
    element: "#option-1",
    popover: {
      className: "first-step-popover-class",
      title: "Title on Popover",
      description: "Body of the popover",
      position: "right",
    },
  },
  {
    element: "#option-3",
    popover: {
      title: "Title on Popover",
      description: "Body of the popover",
      position: "top",
    },
  },
  {
    element: "#option-4",
    popover: {
      title: "Title on Popover",
      description: "Body of the popover",
      position: "left",
    },
  },
  {
    element: "#option-4",
    popover: {
      title: "Title on Popover",
      description: "Body of the popover",
      position: "left",
    },
  }, */
];
