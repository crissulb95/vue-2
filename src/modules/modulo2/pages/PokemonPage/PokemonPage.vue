<template>
  <loader v-if="!pokemon" />
  <div v-else class="pokemon-page">
    <h2>¿Quién es este pokemon?</h2>
    <PokemonPicture :poke-id="pokemon.id" :show="show" />
    <PokemonOptions
      @selection="checkAnswer"
      @newGame="newGame"
      :pokemons="pokemonsArr"
      :correct-pokemon="pokemon"
      :show="show"
    />
  </div>
</template>

<script>
import { PokemonOptions, PokemonPicture } from '@/components/Pokemon'
import { getPokemonOptions } from '@/helpers/pokemon/getPokemon'
import { Loader } from '@/components/Loader'

export default {
  components: { PokemonPicture, PokemonOptions, Loader },
  name: 'PokemonPage',
  data() {
    return {
      pokemonsArr: [],
      pokemon: null,
      show: false,
      showAnswer: false
    }
  },
  methods: {
    async pokemonMix() {
      this.pokemonsArr = await getPokemonOptions()
      const rnd = Math.floor(Math.random() * 4)
      this.pokemon = this.pokemonsArr[rnd]
    },
    checkAnswer() {
      this.show = true
    },
    newGame() {
      this.pokemonsArr = []
      this.pokemon = null
      this.show = false
      this.showAnswer = false
      this.pokemonMix()
    }
  },
  mounted() {
    this.pokemonMix()
  }
}
</script>
<style scoped>
.pokemon-page {
  display: flex;
  flex-wrap: wrap;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: flex-start;
}
.pokemon-page .pokemon-picture-container,
.pokemon-page .pokemon-options-container {
  flex: 0 0 100%;
  width: 100%;
}
</style>
