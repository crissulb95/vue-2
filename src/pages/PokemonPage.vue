<template>
  <loader v-if="!pokemon" />
  <div v-else class="pokemon-page">
    <h2>¬øQui√©n es este pokemon?</h2>
    <PokemonPicture :poke-id="pokemon.id" :show="show" />
    <PokemonOptions
      @selection="checkAnswer"
      @newGame="newGame"
      :pokemons="pokemonsArr"
      :correct-pokemon="pokemon"
      :show="show"
    />
  </div>
</template>

<script>
import { PokemonOptions, PokemonPicture } from "../components/Pokemon";
import { getPokemonOptions } from "@/helpers/pokemon/getPokemon";
import Loader from "../components/Pokemon/Loader.vue";
import { ProductTourJS } from "@/lib/Product Tour/lib.js";
import "@/lib/Product Tour/lib.css";
//import Tour from "@/lib/tour";
//import "@/lib/tour/bootstrap-tour/build/css/bootstrap-tour-standalone.css";
//import introJs from "intro.js";
//console.log(getPokemonOptions());
//console.log("test", Tour);
export default {
  components: { PokemonPicture, PokemonOptions, Loader },
  name: "pokemon-page",
  data() {
    return {
      pokemonsArr: [],
      pokemon: null,
      show: false,
      showAnswer: false,
    };
  },
  methods: {
    async pokemonMix() {
      this.pokemonsArr = await getPokemonOptions();
      const rnd = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonsArr[rnd];
    },
    checkAnswer() {
      /* if (value === this.pokemon.id) {
        this.show = true;
      } */

      this.show = true;
    },
    newGame() {
      this.pokemonsArr = [];
      this.pokemon = null;
      this.show = false;
      this.showAnswer = false;
      this.pokemonMix();
    },
  },
  mounted() {
    this.pokemonMix();

    /*setTimeout(() => {
       introJs()
        .setOptions({
          steps: [
            {
              title: "Welcome",
              intro: "Hello World! üëã",
            },
            {
              element: document.querySelector(".hidden-pokemon"),
              intro: "This step focuses on this hidden image",
              position: "top",
            },
            {
              element: document.querySelector(
                ".pokemon-options-container #option-3"
              ),
              intro: "This step focuses on this option and position",
              position: "right",
            },
            {
              title: "Farewell!",
              element: document.querySelector(".pokemon-options-container"),
              intro: "And this is our final step!",
            },
          ],
        })
        .start(); 
    }, 2000);*/
    /* var tour = new Tour({
      steps: [
        {
          element: "#hidden-pokemon",
          title: "<h4>Title of my step</h4>",
          content: "Content of my step",
          placement: "auto",
        },
        {
          element: ".pokemon-options-container #option-3",
          title: "<h4>Title of next step</h4>",
          content: "Content of next step, searching for option",
          placement: "auto",
          backdrop: true,
          backdropPadding: 15,
        },
      ],
      storage: false,
    });

    setTimeout(() => {
      // Initialize the tour
      tour.init();
      // Start the tour
      tour.start();
      console.log("inside timeout", tour);
    }, 3000);
    console.log("test", tour); */
    const tourOptions = {
      options: {
        darkLayerPersistence: true,
        next: "Next",
        prev: "Previous",
        finish: "Okay!",
        mobileThreshold: 768,
      },

      tips: [
        {
          title: "ü§òüèº My title",
          description: "<h2>My description</h2>",
          selector: ".pokemon-options-container #option-3",
          x: 100,
          y: 50,
          offx: 0,
          offy: 0,
          position: "right",
          onSelected: false,
        },
        {
          title: "Another title",
          description:
            "<h2>My description</h2> <p>this is another description</p>",
          selector: ".pokemon-picture-container #hidden-pokemon",
          x: 50,
          y: 100,
          offx: 0,
          offy: 0,
          position: "bottom",
          onSelected: false,
        },
      ],
    };

    setTimeout(() => {
      ProductTourJS.init(tourOptions);
      ProductTourJS.start();
      console.log("object");
    }, 2000);
  },
};
</script>
<style scoped>
.pokemon-page {
  display: flex;
  flex-wrap: wrap;
  min-height: 100vh;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: flex-start;
  padding: 15px;
}
.pokemon-page .pokemon-picture-container,
.pokemon-page .pokemon-options-container {
  flex: 0 0 100%;
  width: 100%;
}
</style>
