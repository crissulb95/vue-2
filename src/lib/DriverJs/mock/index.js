import { mockArray } from "./mockArray";

export { mockArray };
