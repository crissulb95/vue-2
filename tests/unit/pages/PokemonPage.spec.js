import { mount, shallowMount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage";
import { pokemons } from "../components/Pokemon/pokemons.mock";

describe("PokemonPage Component", () => {
  let wrapper;
  let pokemonMixSpy;

  beforeEach(() => {
    pokemonMixSpy = jest.spyOn(PokemonPage.methods, "pokemonMix");
    wrapper = shallowMount(PokemonPage);
  });

  test("Debe corresponder con el snapshot", () => {
    //console.log(wrapper.html());
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe llamar función pokemonMix al hacer el Mount", () => {
    //El mount ocurre con el shallowMount por lo que llama al espía después de haber invocado a la función,
    //así que no ocurrirá nada después de haber creado el espía,
    //por lo que es necesario hacerlo antes de que se haga el Mount en el beforeEach
    //const pokemonMixSpy = jest.spyOn(PokemonPage.methods, "pokemonMix");
    expect(pokemonMixSpy).toHaveBeenCalled();
  });

  test("Debe corresponder el snapshot cuando carga la data correcta", () => {
    let wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonsArr: [...pokemons],
          pokemon: { name: "cyndaquil", id: 155 },
          show: false,
          showAnswer: false,
        };
      },
    });
    //console.log(wrapper.html());
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe mostrar los componentes PokemonPicture y PokemonOptions", () => {
    let wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonsArr: [...pokemons],
          pokemon: { name: "cyndaquil", id: 155 },
          show: false,
          showAnswer: false,
        };
      },
    });
    const PokemonPicture = wrapper.find("pokemon-picture-stub");
    const PokemonOptions = wrapper.find("pokemon-options-stub");
    //console.log(PokemonPicture.html());
    //console.log(PokemonOptions.html());
    expect(PokemonPicture.exists()).toBeTruthy();
    expect(PokemonOptions.exists()).toBeTruthy();

    //console.log(PokemonPicture.attributes());
    expect(PokemonPicture.attributes("pokeid")).toBe("155");
    expect(PokemonOptions.attributes("pokemons")).toBeTruthy();
    /* Con mount, los attributes como pokeid o show no se van a mostrar, solo los
    típicos como las classes o id, se pueden extraer si se utiliza el shallowMount, 
    y se usan los componentes con sus etiquetas con el sufijo stub que aparece en el 
    snapshot */
  });

  test("Debe desencadenar distintas reacciones al llamar la función checkAnswer", async () => {
    let wrapper = mount(PokemonPage, {
      data() {
        return {
          pokemonsArr: [...pokemons],
          pokemon: { name: "cyndaquil", id: 155 },
          show: false,
          showAnswer: false,
        };
      },
    });

    const PokemonPicture = wrapper.find(".pokemon-picture-container");
    const PokemonOptions = wrapper.find(".pokemon-options-container");

    await wrapper.vm.checkAnswer();

    /* //shallowMount
    expect(PokemonPicture.attributes("show")).toBe("true");
    expect(PokemonOptions.attributes("show")).toBe("true"); */

    const resultImage = PokemonPicture.find("#result");
    const message = PokemonOptions.find(".message");
    //console.log(resultImage.html(), message.text());
    expect(resultImage.exists()).toBeTruthy();
    expect(message.text()).toBe("Es cyndaquil !");

    const correctAnswer = wrapper.findAll("li")[1];
    expect(correctAnswer.attributes("class").includes("correct")).toBeTruthy();

    //console.log(wrapper.findAll("li")[1].html());
  });
});
