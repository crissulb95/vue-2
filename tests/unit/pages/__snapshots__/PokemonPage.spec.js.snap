// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Debe corresponder con el snapshot 1`] = `<loader-stub></loader-stub>`;

exports[`PokemonPage Component Debe corresponder el snapshot cuando carga la data correcta 1`] = `
<div class="pokemon-page">
  <h2>¿Quién es este pokemon?</h2>
  <pokemon-picture-stub pokeid="155" show="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]" correctpokemon="[object Object]" show="false"></pokemon-options-stub>
</div>
`;
